### PR TITLE
feat: add rawHeaders to IncomingMessage

### PR DESCRIPTION
**Pull request by MRayermannMSFT**
#### Description of Change
The goal of this change is to add a `rawHeaders` to `IncomingMessage`, similar to how Node's `http.IncomingMessage` [has](https://nodejs.org/api/http.html#messagerawheaders) a `rawHeaders`. Related to #31531. (Note: link to Node docs may not scroll to the right spot)

I am initially opening this PR as a draft in order to get review and approval.

cc @electron/wg-api 

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added rawHeaders to IncomingMessage.

### CODE REVIEW

**Comment by MRayermannMSFT** [comment]
@ckerr , FYI, I attempted to tag @electron/wg-api as suggested but I think I'm not able to do so since I'm not on the electron team. Feel free to edit my description with a working tag, or just as a comment. Thanks!

File: docs/api/incoming-message.md [hunk 1]
```diff
@@ -80,3 +80,11 @@ An `Integer` indicating the HTTP protocol major version number.
 An `Integer` indicating the HTTP protocol minor version number.
 
 [event-emitter]: https://nodejs.org/api/events.html#events_class_eventemitter
+
+### `response.rawHeaders`
```
Comment (codebytere) [inline docs/api/incoming-message.md:84]: It looks like this maps directly to what's in the [Node.js](https://nodejs.org/api/http.html#messagerawheaders) docs - I'd like to see an example here but generally speaking given that this improves alignment between Node.js and Electron i think it'd be fine to proceed with this as-is!

**Comment by MRayermannMSFT** [comment]
Turns out that is not where the lowercasing is being done. I'll share more about what I found once I have pushed some changes. Also, I am going to go with option 2.

File: docs/api/incoming-message.md (hunk 1 above)
Comment (MRayermannMSFT) [reply docs/api/incoming-message.md:84]: I've added the same example as found in the Node.js docs.

**Comment by MRayermannMSFT** [comment]
So turns out the lowercasing was being done in `electron_api_url_loader.cc`. The converter in `net_converter.cc` seems to potentially be dead code, but I don't know enough to say for sure.

File: spec-main/api-net-spec.ts [hunk 2]
```diff
@@ -1626,6 +1631,137 @@ describe('net module', () => {
...
+        response.statusMessage = 'OK';
+        response.setHeader('last-modified', ['yesterday', 'today']);
+        response.end();
+      });
+      const urlRequest = net.request(serverUrl);
+      const response = await getResponse(urlRequest);
+      expect(response.statusCode).to.equal(200);
+      expect(response.statusMessage).to.equal('OK');
+
+      const headers = response.headers;
```
Comment (MRayermannMSFT) [inline spec-main/api-net-spec.ts:1646]: Let me know if it'd be better to have tests not so explicitly related to `IncomingMessage.rawHeaders` be a different PR that is merged before/after/concurrently to this.

File: lib/browser/api/net.ts [hunk 3]
```diff
@@ -61,31 +61,41 @@ class IncomingMessage extends Readable {
...
-            filteredHeaders[header.key] = header.value;
+            filteredHeaders[keyLowerCase] = header.value;
           }
         }
       }
     });
     return filteredHeaders;
   }
 
+  get rawHeaders () {
```
Comment (MRayermannMSFT) [inline lib/browser/api/net.ts:90]: So far I can't seem to figure out how to add this to the typings for `IncomingMessage`, which in turn seems to be causing builds to fail. From looking at other PRs it looks like perhaps the typing are in another repo and need to be updated? Would love some explicit help here.

**Comment by MRayermannMSFT** [comment]
I think this is now ready for review? Only thing left it seems is to update typings for `IncomingMessage` which I'm not 100% sure on how to do. Am willing to do it of course assuming instructions can be shared. Thanks!

**Comment by MRayermannMSFT** [comment]
@codebytere @ckerr any tips on how to get the electron.d.ts updated so my CI/CD builds can pass?

**Comment by zcbenz** [comment]
Just a notice that this organization won't have much activity this month, so review might have to wait until next month.

**Comment by MRayermannMSFT** [comment]
@zcbenz thanks for letting me know. Happy holidays to you and all the other maintainers!

File: docs/api/incoming-message.md [hunk 4]
```diff
@@ -80,3 +80,25 @@ An `Integer` indicating the HTTP protocol major version number.
...
```
Comment (jkleinsc) [inline docs/api/incoming-message.md:84]:
```suggestion
#### `response.rawHeaders`
```

**Review by jkleinsc — CHANGES_REQUESTED** [review]
@MRayermannMSFT the typings get automatically generated from the documentation.  I think the typing didn't get generated properly because the markdown needed an additional `#`.

**Review by jkleinsc — APPROVED** [review]
API LGTM

File: docs/api/incoming-message.md [hunk 5]
```diff
@@ -80,3 +80,25 @@ An `Integer` indicating the HTTP protocol major version number.
...
 
 [event-emitter]: https://nodejs.org/api/events.html#events_class_eventemitter
+
+#### `response.rawHeaders`
+
+A `string[]` containing the raw HTTP response headers exactly as they were
+received. The keys and values are in the same list. It is not a list of
+tuples. So, the even-numbered offsets are key values, and the odd-numbered
+offsets are the associated values. Header names are not lowercased, and
+duplicates are not merged.
```
Comment (nornagon) [inline docs/api/incoming-message.md:90]: This seems like a weird design choice. Why not a list of tuples?

Comment (MRayermannMSFT) [reply docs/api/incoming-message.md:90]: Because that's what the equivalent thing in the Node.js HTTP/HTTPS module returns: https://nodejs.org/api/http.html#messagerawheaders

I don't mind doing something else if y'all want to diverge, but it was my understanding that the `net` module was supposed to stay fairly similar to the matching Node.js APIs.

Comment (nornagon) [reply docs/api/incoming-message.md:90]: huh, weird. okay, I'm fine with aping Node.js here then.

**Comment by nornagon** [comment]
API LGTM
